src/C++/FieldConvertors.h

```diff
@@ -428,6 +428,11 @@ struct UtcTimeStampConvertor
     char result[ 17+10 ]; // Maximum
     int year, month, day, hour, minute, second, fraction;
 
+    if (precision < 0)
```

Comment:
This looks like a clamp. We don't have std::clamp because we are on an older C++ standard, but I think creating a clamp helper will make this more succinct and reusable.

Reply:
Yeah, it is a clamp indeed. A helper function is added to remove such code duplicate.